### ISSUE TITLE
Modal: Fix and standardize the Close Button label

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#47491]
hte### Description

The `Modal` dialog component accepts a [closeButtonLabel](https://github.com/WordPress/gutenberg/blob/7fb41fc8455804298683495c2053a958c7258d31/packages/components/src/modal/index.tsx#L205) prop that is used for the `aria-label` and tooltip of the Close button.

There are several occurrences of the Modal usage where the passed prop name is, incorrectly, `closeLabel`. It should be `closeButtonLabel`. All the strings passed via `closeLabel` don't have any effect and the default string `Close dialog` is actually rendered.

For example:

The Lock Block modal passes `closeLabel={ __( 'Close' ) }`
The actually rendered label is the default 'Close dialog':

<img width="486" alt="Screenshot 2023-01-27 at 12 24 34" src="https://user-images.githubusercontent.com/1682452/215075123-ab60516e-8c83-443e-80e7-6fb79b58ee1a.png">

Same goes for the Query Loop block > 'Choose a pattern' modal: `closeLabel={ __( 'Cancel' ) }`
The actually rendered label is the default 'Close dialog':

The Preferences modal: `closeLabel={ __( 'Close' ) }`
The actually rendered label is the default 'Close dialog':

etc. I found a total of 18 occurrences of a `closeLabel` prop passed to a `Modal`.

However, instead of fixing the prop name, I'd like to propose to just remove the passed props and simplify the default from 'Close dialog' to 'Close':
- The X close button belongs to the modal dialog. I'm not sure using a label like 'Cancel' is appropriate. that belongs more to the _action flow_ the modal content provides to users.
- The wrong prop names are in place since.a while. Custom labels like 'Cancel' never worked and apparently users didn't suffer that much.
- Simplifying the label to 'Close' may help speech recognition software users to issue more reliable voice commands.

### Step-by-step reproduction instructions

Example test case:
- Edit a post.
- Select a block. In the block toolbar choose: Options > Lock.
- In the Lock block modal dialog, hover the X close button.
- See the tooltip text is 'Close dialog', which is the default from the Modal component.
- Check the code and see the intent was to pass a label text 'Close': https://github.com/WordPress/gutenberg/blob/7fb41fc8455804298683495c2053a958c7258d31/packages/block-editor/src/components/block-lock/modal.js#L91

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
I like the idea of using the better default 👍
